test/functorch/test_eager_transforms.py

```diff
@@ -1295,6 +1295,136 @@ def f(x):
         x = torch.randn(2, 3)
         vmap(f)(x)
 
+    @_set_autograd_function_extension_enabled()
+    def test_none_returns(self, device):
+        class Zeros(torch.autograd.Function):
+            @staticmethod
+            def forward(input):
+                return torch.zeros(input.shape, device=input.device)
+
+            @staticmethod
+            def setup_context(ctx, inputs, output):
+                pass
+
+            @staticmethod
+            def vmap(info, in_dims, input):
+                assert in_dims == (0,)
+                return torch.zeros(input.shape[1:], device=input.device), None
```

Comment:
Why does this work? Does _broadcast_to_and_flatten do something special if out_dims is None? Otherwise it seems that the wrap_fn we define should just not wrap the output in a batched tensor.

Reply:
Your reading is correct -- `_broadcast_to_and_flatten ` doesn't have any special logic for None. It is intended that vmap does not construct a BatchedTensor in this case. 

The expected behavior of this is that the output (call it `zeros`) is not wrapped in a BatchedTensor. That's OK. Inside of `vmap`, a Tensor is either a BatchedTensor or a regular Tensor. This is the latter case.

There are two cases:
- Eventually the regular Tensor interacts with a BatchedTensor. E.g. `regular_tensor * batched_tensor`. This produces a BatchedTensor.
- The regular Tensor gets returned from the function being vmapped over. vmap has special handling to take this Tensor and expand out the batch dimension.